src/components/SearchPanel.vue

```diff
@@ -259,31 +253,28 @@
 
           // loop and update dependants
           await (Promise.allSettled(deps.map(async d => {
-            //create a filter data
-            //It uses to create filter for server request data and as key to of cache
-            const filter_data = getDataForSearchInput.field({
+
+            // cache server data by filter (eg: "zone|eq|A")
+            const filter = getDataForSearchInput.field({
               state,
               field,
               fields: [SEARCH_ALLVALUE, undefined].includes(value) ? [] : [createSingleFieldParameter({ field, value, operator: input.operator })]
             });
 
-            const cached = d.dvalues[filter_data];
+            const cached = d.dvalues[filter];
 
-            d.value  = has_select(d) ? SEARCH_ALLVALUE : null;
-            d.values = Array.from(new Set([                                 // ensure uniques values
+            d.value  = 'selectfield' === d.type ? SEARCH_ALLVALUE : null;
+            d.values = Array.from(new Set([                                       // ensure uniques values
               ...(!has_autocomplete(d) && !is_empty(value) ? [d.values[0]] : []), // get first value (ALL_VALUE)
               ...(!has_autocomplete(d) && is_empty(value) ? d._values      : []), // parent has an empty value (eg. ALL_VALUE) → show all original values on subscriber
-              ...(cached || []),                                                  //cached
+              ...(cached || []),                                                  // cached
             ]));
 
             // value is empty → disable dependants inputs
             d.disabled = is_empty(value) ? d.dependance_strict : false;
 
-            //Check in deep dependencies
-            if (this.state.forminputs
-              .filter(i => i.dependance === d.attribute && (has_select(i) || has_autocomplete(i)))
-              .length > 0
-            ) {
```

Comment:
@volterra79 double check you too, but they should be equivalent.

Reply:
@Raruto i think the problem of deep dependecies is due only if input is manage by select 2 directive. ChangeInput is called if change select input type, but not if it changed dynamically